core/lib/libtomcrypt/src/mpa_desc.c

```diff
@@ -489,7 +489,9 @@ static int montgomery_reduce(void *a, void *b, void *c)
 	LTC_ARGCHK(b != NULL);
 	LTC_ARGCHK(c != NULL);
 	mpanum tmp;
-	init((void **)&tmp);
+	if (init((void **)&tmp) != CRYPT_OK) {
+		return CRYPT_MEM;
```

Comment:
I agree in principle, but in other places it's done like this so I thought I would just do the same.

https://github.com/OP-TEE/optee_os/blob/de57ee09d25a/core/lib/libtomcrypt/src/mpa_desc.c#L83
https://github.com/OP-TEE/optee_os/blob/de57ee09d25a/core/lib/libtomcrypt/src/mpa_desc.c#L360
https://github.com/OP-TEE/optee_os/blob/de57ee09d25a/core/lib/libtomcrypt/src/mpa_desc.c#L391

I can change everywhere if you prefer? But since `init()` is static, in the same file, I would say it's OK to return `CRYPT_MEM` -- we know this is the only possible error and it saves a temporary variable.